[ERC20 Launchpad] - UI for Launch token step

mzparacha opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8705

## Description of Changes
- Added UI for Launch token step of tokenized community creation flow

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_TOKENIZED_COMMUNITY`
- Visit `/createTokenCommunity`
- Fill the form - verify the form validation works as expected
- Submit
- Verify you get redirected to step 2 "Community Information" with a "Not Implemented" label

## Deployment Plan
N/A

## Other Considerations
N/A

masvelio commented on packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:
```diff
@@ -0,0 +1,151 @@
+  };
+
+  const handleCancel = () => {
+    trackAnalytics({
+      event: MixpanelCommunityCreationEvent.CREATE_TOKEN_COMMUNITY_CANCELLED,
+      isPWA: isAddedToHomeScreen,
+    });
```
this should be called if "yes" button is clicked I guess

mzparacha replied:
ah yes, fixed.

masvelio commented on packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:
```diff
@@ -0,0 +1,151 @@
+
+  const { isAddedToHomeScreen } = useAppStatus();
+
+  const [baseOption] = communityTypeOptions;
```
in erc20 weighted topics and in stake we support multiple chains. 

![image](https://github.com/user-attachments/assets/ccd80ef9-5788-4598-8a62-6527b6a0a262)







Do we support only Base for launchpad?

masvelio commented on packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/validation.ts:
```diff
@@ -0,0 +1,24 @@
+  tokenTicker: z
+    .string({ invalid_type_error: VALIDATION_MESSAGES.NO_INPUT })
+    .nonempty({ message: VALIDATION_MESSAGES.NO_INPUT })
+    .max(255, { message: VALIDATION_MESSAGES.MAX_CHAR_LIMIT_REACHED }),
```
I think it should be 3-6 letters according to the ticket. Let's recheck the validations

mzparacha replied:
thanks for the spot, will fix

masvelio commented on packages/commonwealth/client/scripts/views/pages/LaunchToken/utils.ts:
```diff
@@ -0,0 +1,67 @@
+      );
+      return;
+  }
+};
```
those two guys become our little framework for multi-page flows :P

masvelio reviewed (approved):
1. Adjust breadcrumb 


![image](https://github.com/user-attachments/assets/3e1c99c3-ca7b-4565-9ae1-b2d1c5016942)

2. This button seems to be too hight vs design
![image](https://github.com/user-attachments/assets/3e681a2c-13d4-4617-bc73-ac8a31072f39)

3. There are validation requirements in the ticket that do not seem to work properly regarding token name, ticker, description . Will it be implemented further?
![image](https://github.com/user-attachments/assets/03650801-515e-4904-b486-2ed1af1ed166)

4. Mobile does not look good
![image](https://github.com/user-attachments/assets/5b8dff41-9525-438e-8987-81a341b838bd)


Some small comments but looks good to me. Let's fix it and g2g

mzparacha commented:
Thanks for the feedback @masvelio - re https://github.com/hicommonwealth/commonwealth/pull/9193#pullrequestreview-2300193574

1. I thought of adding custom breadcrumb support to fix this and a similar case in https://github.com/hicommonwealth/commonwealth/pull/9194. Created a ticket -> https://github.com/hicommonwealth/commonwealth/issues/9197
2. Actually, I kept it intentionally - some more elements in the new designs didn't follow the existing design system like a Banner and some spacings, for consistency I kept the same styles as we have for normal community creation flow. - lmk if you think we could do something else here.
3. I missed this, Fixed now.
4. Fixed.
